primitives/core/src/assertion/mod.rs

```diff
@@ -189,4 +189,19 @@ impl Assertion {
 			Self::Dynamic(_) => all_web3networks(),
 		}
 	}
+
+	// Given an assertion, normally either "web2" or "web3" identities are expected to be used
+	// to build the assertion, but not both.
+	// However, there's one exception: A1, which requires at least 1 web2 identity and 1 web3
+	// identity.
+	//
+	// This fn is used in `get_eligible_identities` to decide if we should keep web2 identities
+	// regardless of the value of `get_supported_web3networks`
+	#[allow(clippy::match_like_matches_macro)]
+	pub fn force_retain_web2_identity(&self) -> bool {
+		match self {
+			Self::A1 => true,
+			_ => false,
```

Comment:
Can we always keep web2 identities and eventually filter them out at assertion logic level ?

Reply:
Yes thought of those too - but then we'll need to handle that in all 30+ assertions 😂 
It's duplicate code (even if it's just fn call) and easy to forget.

And when I think of relations assertion <> identity. I think most (currently 99%) assertions only need one type of identiies

Are you asking this on behalf of dynamic assertions? There we'll need to handle them all manually anyway I think

Reply:
Yes, but if I understand correctly this will break Dynamic A1 assertion 😿

Reply:
It won't break dynamic A1 tho - it should still work as usual.

It's just the identities are already pre-processed before being passed into the assertion building, and this pre-processing is out of the scope of the contract. But again, A1 is really an edge case 😂